server/src/test/java/org/apache/iotdb/db/engine/compaction/CompactionTaskManagerTest.java

```diff
@@ -337,7 +339,9 @@ public void testRewriteCrossCompactionFileStatus() throws Exception {
     }
 
     CompactionTaskManager.getInstance().submitTaskFromTaskQueue();
-    CompactionTaskManager.getInstance().waitAllCompactionFinish();
+    while (!task.isTaskFinished()) {
+      TimeUnit.MILLISECONDS.sleep(200);
+    }
```

Comment:
Add a failure after some timeout to avoid the test being blocked forever.

Reply:
resolve